libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/NavigationHelperTest.java

```diff
@@ -89,29 +101,40 @@ public void increaseIndex_stepIndexResetsOnLegIndexIncrease() throws Exception {
     assertEquals(0, newIndices.stepIndex());
   }
 
-  //  @Test
-  //  public void checkMilestones_onlyTriggeredMilestonesGetReturned() throws Exception {
-  //    RouteProgress routeProgress = routeProgressBuilder
-  //      .legIndex(0)
-  //      .stepIndex(0)
-  //      .build();
-  //    MapboxNavigationOptions options = MapboxNavigationOptions.builder()
-  //      .defaultMilestonesEnabled(false).build();
-  //    MapboxNavigation mapboxNavigation = new MapboxNavigation(
-  //      RuntimeEnvironment.application.getApplicationContext(), "pk.XXX", options);
-  //    mapboxNavigation.addMilestone(new StepMilestone.Builder()
-  //      .setTrigger(Trigger.eq(TriggerProperty.STEP_INDEX, 0))
-  //      .setIdentifier(1001).build());
-  //    mapboxNavigation.addMilestone(new StepMilestone.Builder()
-  //      .setTrigger(Trigger.eq(TriggerProperty.STEP_INDEX, 4))
-  //      .setIdentifier(1002).build());
-  //
-  //    List<Milestone> triggeredMilestones
-  //      = checkMilestones(routeProgress, routeProgress, mapboxNavigation);
-  //    assertEquals(1, triggeredMilestones.size());
-  //    assertEquals(1001, triggeredMilestones.get(0).getIdentifier());
-  //    assertNotSame(1002, triggeredMilestones.get(0).getIdentifier());
-  //  }
+  @Test
+  public void checkMilestones_onlyTriggeredMilestonesGetReturned() throws Exception {
+    RouteProgress routeProgress = routeProgressBuilder
+      .legIndex(0)
+      .stepIndex(0)
+      .build();
+    MapboxNavigationOptions options = MapboxNavigationOptions.builder()
+      .defaultMilestonesEnabled(false).build();
+    MapboxNavigation mapboxNavigation = new MapboxNavigation(mock(Context.class), ACCESS_TOKEN, options,
+      mock(NavigationTelemetry.class), mock(LocationEngine.class));
+    mapboxNavigation.addMilestone(new StepMilestone.Builder()
+      .setTrigger(Trigger.eq(TriggerProperty.STEP_INDEX, 0))
+      .setIdentifier(1001).build());
+    mapboxNavigation.addMilestone(new StepMilestone.Builder()
+      .setTrigger(Trigger.eq(TriggerProperty.STEP_INDEX, 4))
+      .setIdentifier(1002).build());
+
+    List<Milestone> triggeredMilestones
+      = checkMilestones(routeProgress, routeProgress, mapboxNavigation);
+    assertEquals(1, triggeredMilestones.size());
```

Comment:
Generally is preferable having only 1 assertion per test 👀 https://speakerdeck.com/guardiola31337/elegant-unit-testing-mobilization-2016?slide=57